### PR TITLE
Replace cigarette smoke with particles, add a big puff when taking cigarettes out of your mouth

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Cigarette particles are now particles rather than baked in 
- Taking a cigarette out of your mouth release a big puff of smoke

https://github.com/tgstation/tgstation/assets/51863163/183b39e7-3367-43a1-b750-99e66838d60f

## Why It's Good For The Game

Looks better

## Changelog

:cl: Melbert
add: Cigarette smoke is now more smokey. 
add: Taking a cigarette out of your mouth will let out a big puff of smoke. 
/:cl:

### CODE REVIEW

**Comment by Helg2** [comment]
Can you spam it?

**Comment by RedBaronFlyer** [comment]
now we need a smoking skill that makes you smoke out increasingly obnoxious levels of smoke as you rank up higher and higher

**Comment by atlasle** [comment]
> now we need a smoking skill that makes you blow out increasingly obnoxious levels of smoke as you rank up higher and higher

wizards blowing ship shaped smoke like gandalf image

**Comment by atlasle** [comment]
the smoke cloud is just a sprite getting bigger right? wouldnt be hard to implement some rare alternatives

**Comment by MrMelbert** [comment]
> Can you spam it?

no

> the smoke cloud is just a sprite getting bigger right? wouldnt be hard to implement some rare alternatives

very easy, particles pick from a weighted pool of sprites. so you'd just add a sprite with a very low weight

**Comment by atlasle** [comment]
> very easy, particles pick from a weighted pool of sprites. so you'd just add a sprite with a very low weight
> 
> https://github.com/tgstation/tgstation/blob/c7ddb279636ed65380f8995083c57a69d5bb222b/code/game/objects/effects/particles/smoke.dm#L40-L41
awesome thank you

**Review by Wallemations — APPROVED** [review]

File: code/game/objects/items/cigs_lighters.dm [hunk 1]
```diff
@@ -184,23 +190,59 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
 	SIGNAL_HANDLER
 	source.apply_status_effect(/datum/status_effect/choke, src, lit, choke_forever ? -1 : rand(25 SECONDS, choke_time_max))
 
+/obj/item/clothing/mask/cigarette/proc/on_mob_dir_change(mob/living/source, old_dir, new_dir)
+	SIGNAL_HANDLER
+	if(isnull(mob_smoke))
+		return
+	update_partile_position(mob_smoke, new_dir)
+
+/obj/item/clothing/mask/cigarette/proc/update_partile_position(obj/effect/abstract/particle_holder/to_edit, new_dir = loc.dir)
```
Comment (san7890) [inline code/game/objects/items/cigs_lighters.dm:231]:
```suggestion
/obj/item/clothing/mask/cigarette/proc/update_particle_position(obj/effect/abstract/particle_holder/to_edit, new_dir = loc.dir)
```

(also needs to be fixed elsewhere)

File: code/game/objects/items/cigs_lighters.dm [hunk 2]
```diff
@@ -184,23 +190,59 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
 
 /obj/item/clothing/mask/cigarette/proc/on_forcesay(mob/living/source)
 	SIGNAL_HANDLER
 	source.apply_status_effect(/datum/status_effect/choke, src, lit, choke_forever ? -1 : rand(25 SECONDS, choke_time_max))
 
+/obj/item/clothing/mask/cigarette/proc/on_mob_dir_change(mob/living/source, old_dir, new_dir)
+	SIGNAL_HANDLER
+	if(isnull(mob_smoke))
+		return
+	update_partile_position(mob_smoke, new_dir)
```
Comment (san7890) [inline code/game/objects/items/cigs_lighters.dm:229]:
```suggestion
	update_particle_position(mob_smoke, new_dir)
```

File: code/game/objects/items/cigs_lighters.dm [hunk 3]
```diff
@@ -315,16 +356,26 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+/obj/item/clothing/mask/cigarette/proc/long_exhale(mob/living/carbon/smoker)
+	smoker.visible_message(
+		span_notice("[smoker] exhales a large cloud of smoke from [src]."),
+		span_notice("You exhale a large cloud of smoke from [src]."),
+	)
+	if(!isturf(smoker.loc))
+		return
+
+	var/obj/effect/abstract/particle_holder/big_smoke = new(smoker.loc, /particles/smoke/cig/big)
+	update_partile_position(big_smoke, smoker.dir)
```
Comment (san7890) [inline code/game/objects/items/cigs_lighters.dm:374]:
```suggestion
	update_particle_position(big_smoke, smoker.dir)
```

File: code/game/objects/items/cigs_lighters.dm [hunk 4]
```diff
@@ -415,6 +473,15 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 /obj/item/clothing/mask/cigarette/get_temperature()
 	return lit * heat
 
+/obj/item/clothing/mask/cigarette/proc/make_mob_smoke(mob/living/smoker)
+	mob_smoke = new(smoker, /particles/smoke/cig)
+	update_partile_position(mob_smoke, smoker.dir)
```
Comment (san7890) [inline code/game/objects/items/cigs_lighters.dm:478]:
```suggestion
	update_particle_position(mob_smoke, smoker.dir)
```

**Review by san7890 — CHANGES_REQUESTED** [review]
spelling mistake

**Comment by Wallemations** [comment]
Graaahhh your minor fix has caused me minor inconvenience by making the bot tell me to re-review it!!!!!

**Review by Wallemations — APPROVED** [review]
partile

**Pull request merged**